src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -1256,6 +1256,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectCapability Include="CrossPlatformExecutable" />
   </ItemGroup>
 
+  <!-- Native AOT compilation -->
+  <ItemGroup>
+    <ProjectCapability Include="NativeAOTProperties" Condition="'$(ShowNativeAOTProperties)' == 'true'"/>
+  </ItemGroup>
+
+  <!-- Native AOT properties should be shown by default for projects targeting .NET 8 or higher, except for class libraries, WPF, WinForms projects -->
+  <PropertyGroup>
+    <ShowNativeAOTProperties Condition="'$(ShowNativeAOTProperties)' == ''
+                           and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
+                           and '$(_TargetFrameworkVersionWithoutV)' >= '8.0'">true</ShowNativeAOTProperties>
+    <ShowNativeAOTProperties Condition="'$(UseWPF)' == 'true'
+                             or '$(UseWindowsForms)' == 'true'
+                             or '$(OutputType)' == 'Library'">false</ShowNativeAOTProperties>
```

Comment:
For `'$(OutputType)' == 'Library'` I believe you can publish a library for AOT and it makes a self-contained native library. See https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net7#build-native-libraries

Reply:
You're right, removing on next commit as we do want it to be on for class libraries.